src/rules/unnecessaryElseRule.ts

```diff
@@ -27,9 +27,16 @@ export class Rule extends Lint.Rules.AbstractRule {
         description: Lint.Utils.dedent`
         Disallows \`else\` blocks following \`if\` blocks ending with a \`break\`, \`continue\`, \`return\`, or \`throw\` statement.`,
         descriptionDetails: "",
-        optionExamples: [true],
-        options: null,
-        optionsDescription: "Not configurable.",
+        optionExamples: [true, [true, { allowElseIf: true }]],
+        options: {
+            type: "object",
+            properties: {
+                allowElseIf: { type: "boolean" },
+            },
+        },
+        optionsDescription: Lint.Utils.dedent`
+            You can optionally specify the option \`"allowElseIf"\` to allow "else if" statements.
```

Comment:
> \`"allowElseIf"\`

Rules typically create an all-caps `const` for these option names to reduce the chance of mispelling them anywhere. For example, `curlyRule.ts` declares `OPTION_AS_NEEDED` and `OPTION_IGNORE_SAME_LINE`. Let's switch to that.

Reply:
Done https://github.com/palantir/tslint/pull/4599/commits/03b2414e52f35c0e035b78ede3496dc04839550e